src/Service/Aggregator/RoleAggregator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Service\Aggregator;
+
+use App\Entity\Aggregated\AggregatedPersonRole;
+use App\Repository\Aggregated\AggregatedPersonRoleRepository;
+use App\Repository\Midata\GroupRepository;
+use App\Repository\Midata\PersonRoleRepository;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RoleAggregator extends WidgetAggregator
+{
+    private const NAME = 'widget.roles';
+
+    private AggregatedPersonRoleRepository $personRoleRepository;
+
+    private PersonRoleRepository $midataPersonRoleRepository;
+
+    private EntityManagerInterface $em;
+
+    public function __construct(
+        EntityManagerInterface $em,
+        GroupRepository $groupRepository,
+        AggregatedPersonRoleRepository $personRoleRepository,
+        PersonRoleRepository $midataPersonRoleRepository
+    ) {
+        parent::__construct($groupRepository);
+
+        $this->em = $em;
+        $this->personRoleRepository = $personRoleRepository;
+        $this->midataPersonRoleRepository = $midataPersonRoleRepository;
+    }
+
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    public function aggregate(DateTime $startDate = null)
+    {
+        // check if People have quit their jobs, and if so aggregate it
+        $listOfUnfinished = $this->personRoleRepository->getUnfinished();
+        foreach ($listOfUnfinished as $unfinished) {
+            $midataObject = $this->midataPersonRoleRepository->find($unfinished->getMidata());
+            $deletedAt = $midataObject->getDeletedAt();
+            if ($unfinished->getId() == 217) {
```

Comment:
@SebastianStorz why is here an id check. wouldn't this make this logic only work for one specific db row? and please replace the magic number with a public constant